### PR TITLE
ref(data-structs): set uses visitors

### PR DESCRIPTION
**Pull request by dudymas**
consider using visitors. it's hard on my eyes when the functions have so many responsibilites... loop manage, instance creation, etc etc.

just opinions, of course.

### CODE REVIEW

File: data-structures/sets/set.go [hunk 1]
```diff
@@ -33,45 +33,41 @@ func (set *Set) Clear() {
...
-		larger = other
-	} else {
-		smaller = other
-		larger = set
-	}
-	for v, _ := range smaller.elements {
-		if larger.Contains(v) {
+	set.visit(func(v interface{}) {
+		if predicate(v) {
 			result.Add(v)
```
Comment (dudymas) [inline data-structures/sets/set.go]: wow, this diff is terrible

**Comment by jasonkeene** [comment]
LGTM and much cleaner!

**Comment by luqmaan** [comment]
:hankey:

**Comment by dudymas** [comment]
@luqmaan aren't you supposed to be working on the Fresno version of that HARTXT thing?

**Comment by dudymas** [comment]
@luqmaan get it? get it?
#terrible-at-jokes

**Comment by luqmaan** [comment]
Hiiii, https://github.com/luqmaan/Instabus

I don't get it tho. :crying_cat_face:

**Comment by dudymas** [comment]
HART is for Houston Area Rapid Transport. If it was fresno... fill in the blank.... _ART

**Comment by luqmaan** [comment]
:dash:

**Comment by theY4Kman** [comment]
This is sexy. I'mma use this shit.

**Pull request merged**